activemq-kahadb-store/src/main/java/org/apache/activemq/store/kahadb/KahaDBStore.java

```diff
@@ -766,6 +773,83 @@ public void execute(Transaction tx) throws Exception {
                             }
                         }
                         sd.orderIndex.stoppedIterating();
+
+                        if(revertOrderIndex && position > 0) {
+                            sd.orderIndex.resetCursorPosition();
```

Comment:
This does not revert the index to the previous location, it reverts it to 0 at the beginning. One way to avoid having side effects and having to reset at all is to create a different MessageOrderCursor and not the internal one as that tracks state.